plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -694,11 +697,13 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                             return Stream.of();
                         }
                         String tableLocation = metastoreTable.get().location();
-                        MetadataEntry metadata = transactionLogAccess.getMetadataEntry(getSnapshot(table, tableLocation, session), session);
+                        TableSnapshot snapshot = getSnapshot(table, tableLocation, session);
+                        MetadataEntry metadata = transactionLogAccess.getMetadataEntry(snapshot, session);
+                        ProtocolEntry protocol = transactionLogAccess.getProtocolEntry(session, snapshot);
```

Comment:
Can this end up doing second time same IO as done just above (in `transactionLogAccess.getMetadataEntry`)?
Looking at the code this may happen if snapshot has a checkpoint attached.

Reply:
Yes, it will increase IO in such cases :(

Reply:
will this be solved after https://github.com/trinodb/trino/pull/18919 ?